### PR TITLE
Update README.md to include plug submodule

### PR DESCRIPTION
**Pull request by forrcaho**
I _think_ this is correct? I was going nuts trying to figure out how Gwion was supposed to actually make sound ...

I'll make the corresponding change in the docs (only) if you confirm this is correct.

(Pretty cheesy for a Hacktoberfest PR, I know ... I'll try to write some actual code once I get Gwion to actually work and stuff.)

### CODE REVIEW

**Comment by fennecdjay** [comment]
> I was going nuts trying to figure out how Gwion was supposed to actually make sound ...

Sorry about that. That's clearly a problem in the documentation.
Currently there are a few usable drivers.
I think the most portable ones are [Jack](https://github.com/fennecdjay/Gwion-plug/tree/master/Jack) and [WebSocket](https://github.com/fennecdjay/Gwion-plug/tree/master/WebSocket), the later needing improvments. Maybe you can mention this somewhere (I guess a `Wait, How do this make sound` section would be nice).

It's fine if you PR something in the docs when we're done with this PR.

However, I you're willing to do more, there are lots of points we can address
(I'm putting these here for later reference, I'm not sure I can update all issues and project before tomorrow)

  * Make the plugins able to parse arguments. (should be simple, it might be that using [this](https://github.com/fennecdjay/Gwion/blob/master/src/plug.c#L173-L184) could works, provided it is not static anymore.

  * Updating some drivers: look in plug/driver_todo

  * Porting STK (this one would be pretty nice to have, and useful for [this](https://github.com/tonal-glyph/gwiuck) project). Probably easier using my [fork](https://github.com/fennecdjay/gwion-swig) of swig.

  * As stated earlier, WebSocket driver needs improvement:
    + using AudioWorklet over ScriptProcessor
    + using wasm to copy input array
    + using TLS.

  * There is also some work to do on closures, but that should be a discussion in the team before, I think. (However, [this link](http://craftinginterpreters.com/closures.html) is pretty nice on the subject.

  * A tutorial on how to create a plugin. Maybe your physmod project could be used here. This is a bit more involved, but I'll be happy to help.

Thank you for your interrest and questions, it helps me clarify the current state of the project.

**Comment by fennecdjay** [comment]
Would you be OK to merge?
We can create other issues for the missing info and updating the docs.

**Comment by forrcaho** [comment]
I'm going to revise my text a bit before I submit it.

**Comment by forrcaho** [comment]
Please take a look at this before I merge. A few issues are (1) it only talks about linux/jack (don't really know about other OSes), and (2) it assumes we can get the soundpipe lib to build as part of the Soundpipe plugin build (I'm assuming here that the Soundpipe module will link with the static soundipipe.a library.

**Comment by forrcaho** [comment]
@fennecdjay It seems I can't acutally merge this myself. I think being on the Gwion project and being a maintainer for this repo are two different things -- another reason to move this repo under the Gwion project. In any case I wanted you to look at it; what I wrote about getting sound is totally linux-centric, and I don't know if we need some OS X language in there as well (although I don't develop on OS X so it would be hard for me to write).

File: README.md
Comment (fennecdjay) [inline README.md:101]: Maybe add something along those lines
```md
> Currently those instructions are Linux-centric, we are working to improve that.
```

**Review by fennecdjay — COMMENTED** [review]
Mayb

**Comment by fennecdjay** [comment]
I suggested some fixes, maybe you can add them and try to merge (I'm not sure yet you have those rights, it'll be nice to check).

This clearly opens the problem of handling submodules/dependencies in the plug submodule. :+1:

File: README.md
Comment (forrcaho) [reply README.md]: Changed as recommended

File: README.md
Comment (forrcaho) [reply README.md]: Added to the top of the "making sound" section.

**Review by fennecdjay — APPROVED** [review]
Looks good to me!

**Comment by fennecdjay** [comment]
Can you merge it?

**Comment by forrcaho** [comment]
It looks like I still don't have the required permissions to merge PRs into this repo.

**Comment by fennecdjay** [comment]
Ok, checking that.

**Comment by fennecdjay** [comment]
Can you merge now?

**Comment by forrcaho** [comment]
Merged!

**Pull request merged**
